core/server/master/src/main/java/alluxio/master/file/options/CompleteFileContext.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0 (the
+ * "License"). You may not use this work except in compliance with the License, which is available
+ * at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.file.options;
+
+import alluxio.grpc.CompleteFilePOptions;
+import alluxio.master.file.FileSystemMasterOptions;
+import alluxio.underfs.UfsStatus;
+import com.google.common.base.MoreObjects;
+
+/**
+ * Wrapper for {@link CompleteFilePOptions} with additional context data.
+ */
+public class CompleteFileContext extends OperationContext<CompleteFilePOptions.Builder> {
+
+  private long mOperationTimeMs;
+  private UfsStatus mUfsStatus;
+
+  // Prevent instantiation
+  private CompleteFileContext() {
+    super(null);
+  };
+
+  /**
+   * Creates rename context with given option data.
+   *
+   * @param mergedOptionsBuilder options builder
+   */
+  private CompleteFileContext(CompleteFilePOptions.Builder mergedOptionsBuilder) {
+    super(mergedOptionsBuilder);
+    mOperationTimeMs = System.currentTimeMillis();
+    mUfsStatus = null;
+  }
+
+  /**
+   * Merges and embeds the given {@link CompleteFilePOptions} with the corresponding master options.
+   *
+   * @param optionsBuilder Builder for proto {@link CompleteFilePOptions} to embed
+   * @return the instance of {@link CompleteFileContext} with default values for master
+   */
+  public static CompleteFileContext defaults(CompleteFilePOptions.Builder optionsBuilder) {
+    CompleteFilePOptions masterOptions = FileSystemMasterOptions.getCompleteFileOptions();
```

Comment:
naming is confusing here - `FileSystemMasterOptions.getCompleteFileOptions()` gets default option values but doesn't mention anything about defaults, while `CompleteFileContext.defaults(builder)` is named `defaults`, but returns non-default values

One idea:
`defaults` -> `applyDefaults`
`FileSystemMasterOptions.getCompleteFileOptions` -> `FileSystemMasterOptions.completeFileDefaults()`

Reply:
we could also move the `FileSystemMasterOptions.getCompleteFileOptions` to be `private static CompleteFileContext.defaults()`. I don't think it needs to be exposed outside of `CompleteFileContext`

Reply:
Ultimate plan is to get rid of FileSystemMasterOptions and move the definitions to corresponding operation contexts. I'll do that all at once for all operation types.

Reply:
After reconsideration I feel like *Context.defaults() overload for taking proto options makes sense. Parameter javadoc also clearly states that given options are due for merging with defaults.

Reply:
I still think it's misleading. We have historically used `.defaults()` methods on all of our option classes to return the default options. How about `CompleteFileContext.create(completeFileOptions)`?